homeassistant/components/weather/openweathermap.py

```diff
@@ -129,10 +135,31 @@ def attribution(self):
         """Return the attribution."""
         return ATTRIBUTION
 
+    @property
+    def forecast(self):
+        """Return the forecast array."""
+        if self._forecast is False:
+            return None
+
+        forecast_array = []
+
+        for forecast_entry in self.forecast_data.get_weathers():
```

Comment:
What is the reference time of a forecast in this context? If you would change to a dict of dicts you would have to be sure that all keys are unique. I think a list of dicts as it is currently looks fine. Try to answer how would this data structure be used and what data does it contain. Then you will know better what kind of structure is appropriate.